syntax = "proto3";

package TW.Ontology.Proto;
option java_package = "wallet.core.jni.proto";

// Input data necessary to create a signed transaction.
message SigningInput {

    string contract = 1;

    string method = 2;

    bytes owner_private_key = 3;

    // base58 encode address string (160-bit number)
    string to_address = 4;

    uint64 amount = 5;

    bytes payer_private_key = 6;

    uint64 gas_price = 7;

    uint64 gas_limit = 8;

    // base58 encode address string (160-bit number)
    string query_address = 9;

    uint32 nonce = 10;

}

// Transaction signing output.
message SigningOutput {
    // Signed and encoded transaction bytes.
    bytes encoded = 1;
}
